projects/igniteui-angular/src/lib/icon/icon.service.ts

```diff
@@ -128,18 +132,40 @@ export class IgxIconService {
         return fontSet + '_' + iconName;
     }
 
+    /**
+   * @hidden
+   */
+    public ngOnDestroy(): void {
+        this.zone.runOutsideAngular(() => {
+            this._httpRequest.removeEventListener('load', this._loadListener);
+        });
+    }
+
     /**
      * @hidden
      */
     private fetchSvg(iconName: string, url: string, fontSet: string = '') {
-        const request = this._httpClient.get(url, { responseType: 'text' });
-        const subscription = request.subscribe((value: string) => {
-            this.cacheSvgIcon(iconName, value, fontSet);
-        }, (error) => {
-            throw new Error(`Could not fetch SVG from url: ${url}; error: ${error.message}`);
-        }, () => {
-            subscription.unsubscribe();
+        this._httpRequest = new XMLHttpRequest();
+        const args = { iconName, url, fontSet };
+        this.zone.runOutsideAngular(() => {
+            this._loadListener = this.onLoad.bind(this, args);
+            this._httpRequest.addEventListener('load', this._loadListener);
```

Comment:
Please use `onLoad` instead of attaching event listener.

Reply:
Event listener removed